hooks/windows/check-arch.js

```diff
@@ -0,0 +1,53 @@
+module.exports = function(ctx) {
+    if (ctx.opts && ctx.opts.platforms && ctx.opts.platforms.indexOf('windows') > -1
+        && ctx.opts.options) {
+        var shell = ctx.requireCordovaModule('shelljs');
+        var path = ctx.requireCordovaModule('path');
+        var nopt = ctx.requireCordovaModule('nopt');
+
+        // parse and validate args
+        var args = nopt({
+            'archs': [String],
+            'appx': String,
+            'phone': Boolean,
+            'win': Boolean,
+            'bundle': Boolean,
+            'packageCertificateKeyFile': String,
+            'packageThumbprint': String,
+            'publisherId': String,
+            'buildConfig': String
+        }, {}, ctx.opts.options.argv, 0);
+
+        // Check if --appx flag is passed so that we have a project build version override:  
+        var isWin10 = args.appx && args.appx.toLowerCase() === 'uap';
+
+        // Else check "windows-target-version" preference:
+        if (!isWin10) {
```

Comment:
Do we have a problems w/ AnyCpu architecture on Windows 8.1/Windows Phone 8.1? Shouldn't we just `return` here?